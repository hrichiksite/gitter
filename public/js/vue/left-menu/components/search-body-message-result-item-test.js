const mount = require('../../__test__/vuex-mount');
const { default: SearchBodyMessageResultItem } = require('./search-body-message-result-item.vue');

const {
  createSerializedRoomFixture,
  createSerializedMessageSearchResultFixture
} = require('../../__test__/fixture-helpers');

describe('search-body-message-result-item', () => {
  // So the dates look the same wherever you actually are
  require('moment-timezone').tz.setDefault('America/Los_Angeles');

  it('matches snapshot', () => {
    const messageSearchResult = createSerializedMessageSearchResultFixture();
    const { wrapper } = mount(SearchBodyMessageResultItem, { messageSearchResult }, store => {
      const room1 = createSerializedRoomFixture('community/room1');
      store.state.roomMap = {
        [room1.id]: room1
      };
      store.state.displayedRoomId = room1.id;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls store action "jumpToMessageId" after clicking result', () => {
    const messageSearchResult = createSerializedMessageSearchResultFixture();
    const { wrapper, stubbedActions } = mount(
      SearchBodyMessageResultItem,
      {
        messageSearchResult
      },
      store => {
        const room1 = createSerializedRoomFixture('community/room1');
        store.state.roomMap = {
          [room1.id]: room1
        };
        store.state.displayedRoomId = room1.id;
      }
    );

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(stubbedActions.jumpToMessageId).toHaveBeenCalled();
  });
});
