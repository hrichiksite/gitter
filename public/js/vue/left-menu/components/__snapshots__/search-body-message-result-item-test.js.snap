// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-body-message-result-item matches snapshot 1`] = `
<li
  class="message-search-item"
>
  <span
    class="message-search-item-detail"
  >
    <a
      class="message-search-item-detail-author-link"
      href="/viktomas_gitlab"
    >
      Tomas Vik
      
      <span>
        @viktomas_gitlab
      </span>
    </a>
     
    <span>
      ‚óè
    </span>
     
    <a
      class="message-search-item-detail-permalink"
      href="/community/room1?at=573bd813e2996a5a42c95899-1"
    >
      May 17th 7:48 PM
    </a>
  </span>
   
  <span
    class="message-search-item-text"
  >
    
    @MadLittleMods For some groups it might not matter much, but for others a lot.  Kinda like stack overflows 'take this offline' thing when you start doing a conversation in the comments.  Im' sure it's not simple to design
  
  </span>
</li>
`;
